src/main/java/org/spongepowered/api/service/scheduler/SchedulerQuery.java

```diff
@@ -26,70 +26,54 @@
 
 import com.google.common.base.Optional;
 
-import java.util.Collection;
+import java.util.Set;
 import java.util.UUID;
 
-
 /**
- * Examples of how to setup the use of the Scheduler are included in the API descriptions in this interface.
- *
- * @see AsynchronousScheduler#runTask(Object, Runnable)
- * @see AsynchronousScheduler#runTaskAfter(Object, Runnable, java.util.concurrent.TimeUnit, long)
- * @see AsynchronousScheduler#runRepeatingTask(Object, Runnable, java.util.concurrent.TimeUnit, long)
- * @see AsynchronousScheduler#runRepeatingTaskAfter(Object, Runnable, java.util.concurrent.TimeUnit, long, long)
- *
- * <p>Examples of how to setup the use of the Scheduler are included in the API descriptions in this interface.</p>
- *
- * @see SynchronousScheduler#runTask(Object, Runnable)
- * @see SynchronousScheduler#runTaskAfter(Object, Runnable, long)
- * @see SynchronousScheduler#runRepeatingTask(Object, Runnable, long)
- * @see SynchronousScheduler#runRepeatingTaskAfter(Object, Runnable, long, long)
+ * Represents a scheduler that can be queried for tasks that it is running or
+ * that are scheduled to run.
  */
-
 public interface SchedulerQuery {
 
     /**
      * Retrieves a scheduled or running task by its unique ID.
      *
      * @param id The id of the task
-     * @return The scheduled or running task, or {@link com.google.common.base.Optional#absent()}
+     * @return The scheduled or running task, or {@link Optional#absent()}
      */
     Optional<Task> getTaskById(UUID id);
 
     /**
-     * Gets the UUID of the task by name.
+     * Gets a set of task IDs for the given task name. There may be more than
+     * one task with the same name.
      *
-     * @param name  The name of the task to search
-     * @return The Optional&lt;UUID&gt; result from the search by name.
+     * @param name The name of the task to search
+     * @return A set of task IDs, empty if no tasks have that name
      */
-    Optional<UUID> getUuidOfTaskByName(String name);
+    Set<UUID> getUuidOfTaskByName(String name);
```

Comment:
Is this method really needed? The method below (task by name) should provide the same